packages/app/pages/submission/new.tsx

```diff
@@ -1,5 +1,137 @@
+import { styled } from "@/stitches.config";
+import debounce from "lodash.debounce";
+import { useState } from "react";
+import {useRouter} from "next/router";
+import { 
+  Button, 
+  Input,
+  Text 
+} from "@cabindao/topo";
+import { useStore } from "@/store/store";
+import Title from "@/components/Title";
+import Post from "@/components/Post";
+import { IPost } from "@/interfaces";
+
+/** Styled components */
+const URLInput = styled(Input, {
+  flex: "1",
+  width: "100%"
+});
+
+const SubmitButton = styled(Button, {
+  marginTop: "$5"
+});
+
+/** Page: 'new' */
 const NewSubmission = () => {
-  return <div>new submission page</div>;
+  const router = useRouter();
+  const { currentProfile } = useStore();
+
+  const [error, setError] = useState<string>('');
+  // TODO: Use real wallet address
+  const [postData, setPostData] = useState<IPost>({
+    _id: '',
+    title: '',
+    domainText: '',
+    url: '',
+    // postedBy: currentProfile.walletAddress,
+    postedBy: '0x00000',
+    created_at: new Date().toISOString(),
+    upvotes: 0
+  });
+
+  /** Fetch post data from the backend */
+  const fetchPostData = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    const url = e.target.value;
+    if (url) {
+      const res = await fetch(`/api/url/metadata`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json"
+        },
+        body: JSON.stringify({
+          url: url
+        })
+      });
+      
+      if (res.status !== 200) {
+        // Show error and reset metadata on invalid input
+        setError('Invalid input, please check the URL and try again.');
+        setPostData({
+          ...postData,
+          title: '',
+          domainText: '',
+          url: ''
+        });
+      } else {
+        // Set meta data for preview
+        const metaData = await res.json();
+        setPostData({
+          ...postData,
+          title: metaData.title || '',
+          domainText: metaData.description || metaData.og_description || '',
+          url: metaData.url
+        });
+        setError('');
+      }
+    }
+  };
+
+  const onInputDebounce = debounce(fetchPostData, 300);
```

Comment:
you'll want to wrap this in a `useMemo` to prevent redefining this on each render